scopes/dependencies/dependency-resolver/get-outdated-workspace-pkgs.ts

```diff
@@ -0,0 +1,118 @@
+import { DependencyResolverAspect } from './dependency-resolver.aspect';
+import { ManifestDependenciesKeysNames } from './manifest';
+import { WorkspacePolicy } from './policy';
+
+type CurrentPkg = {
+  name: string;
+  currentRange: string;
+  source: 'variants' | 'component' | 'rootPolicy';
+  variantPattern?: string | null;
+  componentId?: string;
+  targetField: ManifestDependenciesKeysNames;
+};
+
+export type OutdatedPkg = CurrentPkg & {
+  latestRange: string;
+};
+
+type ResolveFunction = (range: string) => Promise<string | null>;
+
+export function getOutdatedWorkspacePkgs({
```

Comment:
add a comment above this function that says that it get it from the different sources - root policy, variants, component config file